cartography/graph/statement.py

```diff
@@ -59,30 +64,31 @@ def as_dict(self):
             "iterationsize": self.iterationsize,
         }
 
-    def _run(self, session):
+    def _run(self, tx: neo4j.Transaction) -> neo4j.StatementResult:
         """
         Non-iterative statement execution.
         """
-        return session.run(self.query, self.parameters)
+        return tx.run(self.query, self.parameters)
 
-    def _run_iterative(self, session):
+    def _run_iterative(self, tx: neo4j.Transaction) -> None:
         """
         Iterative statement execution.
 
         Expects the query to return the total number of records updated.
         """
         self.parameters["LIMIT_SIZE"] = self.iterationsize
 
-        # TODO improve this
-        done = False
-        while not done:
-            results = self._run(session)
-            done = False
-            for r in results:
-                if int(r['TotalCompleted']) == 0:
-                    done = True
+        total_completed = -1
+        while total_completed != 0:
+            result: neo4j.StatementResult = self._run(tx)
+
+            record: neo4j.Record = result.single()
+            # TODO: use the BoltStatementResultSummary object to determine the number of items processed
```

Comment:
Out of scope for this PR, but it would be great to instead use the built-in BoltStatementResultSummary object and get the # nodes/# rels updated instead of relying on the query writer to do a `MATCH/MERGE ... RETURN ...count(*) as TotalCompleted` just for this logic to work.